tokio/src/process/unix/mod.rs

```diff
@@ -177,8 +177,15 @@ impl AsRawFd for Pipe {
 }
 
 pub(crate) fn convert_to_stdio(io: PollEvented<Pipe>) -> io::Result<Stdio> {
-    io.into_inner()
-        .map(|io| unsafe { Stdio::from_raw_fd(io.fd.into_raw_fd()) })
+    let mut fd = io.into_inner()?.fd;
+
+    // Ensure that the fd to be inherited is set to *blocking* mode, as this
+    // is the default that virtually all programs expect to have. Those
+    // programs that know how to work with nonblocking stdio will know how to
+    // change it to nonblocking mode.
+    set_nonblocking(&mut fd, false)?;
```

Comment:
So this is ok because neither end of the pipe is actually used in Tokio?

Reply:
Correct. One end of the pipe should be owned by the child process (and already closed in the parent). The other end of the pipe is moved into this function, and calling `.into_inner()` unwraps the `PollEvented` and deregisters it from the reactor.